cmake/scripts/common/Macros.cmake

```diff
@@ -76,7 +77,7 @@ function(core_add_library name)
     if(NOT CORE_SYSTEM_NAME STREQUAL windowsstore)
       list(APPEND lib_DEPS dvdnav)
     endif()
-    add_dependencies(${name} ${lib_DEPS})
+    add_dependencies(${name} ${lib_DEPS} ${DEPENDS})
```

Comment:
When I revert this the cmake step fails:

```
CMake Error at xbmc/cores/RetroPlayer/savestates/CMakeLists.txt:16 (add_dependencies):
  Cannot add target-level dependencies to non-existent target
  "retroplayer_savestates".

  The add_dependencies works for top-level logical targets created by the
  add_executable, add_library, or add_custom_target commands.  If you want to
  add file-level dependencies see the DEPENDS option of the add_custom_target
  and add_custom_command commands.
```

I think this is because macros.cmake adds the library, but the savestate CMakeLists.txt tried to add the dependency, which can't see the library target from the macro.

Reply:
In VS (and XCode) there is only one target (xbmc), where everything is added from the `core_add_library` calls. So the one you want to set as dependency doesn't exist.

Reply:
I see. Does the latest fix look OK?